Add support for Electrolux heatpump and bump arduino-heatpumpir version

ikilledmypc opened the pull request:
# What does this implement/fix?

Quick description and explanation of changes

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#1999

## Test Environment

- [X] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266

## Example entry for `config.yaml`:
<!--
  Supplying a configuration snippet, makes it easier for a maintainer to test
  your PR. Furthermore, for new integrations, it gives an impression of how
  the configuration would look like.
  Note: Remove this section if this PR does not have an example entry.
-->

```yaml
# Example config.yaml
remote_transmitter:
  pin: GPIO19
  # Infrared remotes use a 50% carrier signal
  carrier_duty_percent: 50%
  
climate:
  - platform: heatpumpir      
    name: "heatpump home"
    protocol: electroluxyal
    horizontal_default: auto
    vertical_default: auto
    max_temperature: 30
    min_temperature: 17
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

~~I looked at the documentation and I am wondering if I can test this against my existing installation of home assistant as well to make sure this works, to avoid having to setup a separate instance. If someone can help me on how to test this integration I would be very grateful~~
edit: **I just tested it locally. My instance of home assistant automatically recognized the device as new device and I was able to control my thermostat from home assistant**

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).
[#1999](https://github.com/esphome/esphome-docs/pull/1999)

ikilledmypc committed:
Add support for Electrolux heatpump and bump arduino-heatpumpir version

ikilledmypc committed:
fixed formatting heatpumpir.cpp

jesserockz commented on esphome/components/heatpumpir/climate.py:
```diff
@@ -114,4 +115,4 @@ def to_code(config):
     # PIO isn't updating releases, so referencing the release tag directly. See:
     # https://github.com/ToniA/arduino-heatpumpir/commit/0948c619d86407a4e50e8db2f3c193e0576c86fd
-    cg.add_library("", "", "https://github.com/ToniA/arduino-heatpumpir.git#1.0.18")
+    cg.add_library("", "", "https://github.com/ToniA/arduino-heatpumpir.git#1.0.20")
```
While we are making changes, I would like to switch this to the platformio lib
```suggestion
    cg.add_library("tonia/HeatpumpIR", "1.0.20")
```

jesserockz commented on platformio.ini:
```diff
@@ -64,7 +64,7 @@ lib_deps =
     ; PIO isn't update releases correctly, see:
     ; https://github.com/ToniA/arduino-heatpumpir/commit/0948c619d86407a4e50e8db2f3c193e0576c86fd
-    https://github.com/ToniA/arduino-heatpumpir.git#1.0.18  ; heatpumpir
+    https://github.com/ToniA/arduino-heatpumpir.git#1.0.20  ; heatpumpir
```
```suggestion
    tonia/HeatpumpIR@1.0.20  ; heatpumpir
```

ikilledmypc committed:
Update esphome/components/heatpumpir/climate.py

Switch back to platformIO library version of HeatpumpIR for Climate

Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

ikilledmypc committed:
Update platformio.ini

Switch back to platformIO library version of HeatpumpIR instead of direct tag reference

Co-authored-by: Jesse Hills <3060199+jesserockz@users.noreply.github.com>

ikilledmypc committed:
Delete comment no longer relevant for heatpumpIR dep

Delete comment no longer relevant for heatpumpIR dep

ikilledmypc committed:
Delete comment no longer relevant for HeatpumpIR
